Automattic/newspack-plugin
# refactor(ia): audience development setup and tabs

### All Submissions:






### Changes proposed in this Pull Request:

p1732114314318549-slack-C06MPMZMF29

Introduces tabbed navigation for the Audience Development wizard.

The "Advanced Settings" toggle has also been removed and the sections that weren't placed under a tab are always visible below the RAS wizard.

<img width="1371" alt="image" src="https://github.com/user-attachments/assets/e7887d7a-78a3-4deb-ba76-198c656986e6">

Tabs should only become visible after RAS activation.

<img width="1373" alt="image" src="https://github.com/user-attachments/assets/9bf0d403-100f-4e4f-ae0d-d388afb9b9fc">

<img width="1191" alt="image" src="https://github.com/user-attachments/assets/0f277dc3-9aa5-4328-aa68-88180ebc5c0c">

<img width="1192" alt="image" src="https://github.com/user-attachments/assets/4429ad00-1f64-4a60-9e1b-2e7f53b35476">

<img width="1191" alt="image" src="https://github.com/user-attachments/assets/3605b0f0-907a-4e42-b7f5-457e29627d63">

### How to test the changes in this Pull Request:

1. Delete the `newspack_reader_activation_enabled` option (CLI: `wp option delete newspack_reader_activation_enabled`)
2. Make sure you have WooCommerce Memberships available to test the Content Gating tab
3. Visit "Audience -> Setup" and confirm the pending RAS wizard renders without any tabs
4. Complete the setup and confirm you are redirected back but now with all the tabs and additional sections below the RAS setup
5. With WooCommerce Memberships deactivated, confirm the "Content Gating" tab is not available
6. Activate the plugin and confirm the tab is now available
7. Visit "Content Gating" and confirm the "Display memberships on the subscriptions tab" toggle works and the selection persists (refresh the page)
8. Click to Configure your gate and confirm you are redirected to the gate editor
9. Back to the wizard, visit the "Transactional Emails" tab
10. Confirm it renders as the image above
11. Click to edit the emails and confirm you are redirected to the editor

### Other information:
